src/Concerns/ExtractsDatasetKeys.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pest\Concerns;
+
+use Pest\Exceptions\DatasetDoesNotExist;
+
+const DATASET_KEY_EXTRACTION_PATTERN = '/^(.*):(.*)/';
+
+trait ExtractsDatasetKeys
```

Comment:
Nice the methods are much easier to follow now 👍  

Maybe add a few comments in this trait though, I think I would at least like a comment on the trait itself explaining what this trait actually does. At least it might not be that obvious for people what we mean by dataset keys 😄

Reply:
Added a trait comment:

Implements some methods useful to extract specific keys from a registered dataset.

```
/**
 * Implements some methods useful to extract specific keys from a registered dataset.
 *
 * given a dataset: dataset('people',  'Nuno'   => [
 *      'Nuno' => ['name' => 'Nuno', 'country' => 'Portugal', 'role' => 'Owner'],
 *      'Fabio' => ['name' => 'Fabio', 'country' => 'Italy', 'role' => 'Member'],
 *      'Oliver' => ['name' => 'Oliver', 'country' => 'Denmark', 'role' => 'Maintainer'],
 *  ],
 *
 * Keys are selected by using a specific pattern when calling a dataset for a
 * specific test, eg. ->with('people:name,age')
 */
trait ExtractsDatasetKeys
{ ...
```